functions/Get-DbaUserPermission.ps1

```diff
@@ -215,17 +215,65 @@ function Get-DbaUserPermission {
                 $dbs = $dbs | Where-Object IsSystemObject -eq $false
             }
 
-            #reset $serverDT
-            $serverDT = $null
+            # grab server data (just the one time):
+            Write-Message -Level Verbose -Message "Building data table for server objects"
 
-            foreach ($db in $dbs) {
-                Write-Message -Level Verbose -Message "Processing $db on $instance"
+            $sqlFile = Join-Path -Path $script:PSModuleRoot -ChildPath "bin\stig.sql"
+            $sql = [System.IO.File]::ReadAllText("$sqlFile")
+            $sql = $sql.Replace("<TARGETDB>", $master.Name)
 
-                $db.ExecuteNonQuery($endSQL)
+            try {
+                $master.ExecuteNonQuery($sql)
+            } catch {
+                # here to avoid an empty catch
+                $null = 1
```

Comment:
So you don't care if executing the $sql fails or not? The rest of the command would just run fine in that case? Really?

Reply:
The code that existed in the function indicated that this try/catch was here simply becuase it _might_ fail if the objects were not cleaned up from a previous run (where a user may have control c'd their way out of the code. This simply was not part of the function I was trying to fix and it seems to have been working fine to date. If you have something specific you would like to see here please feel free to add it.

Reply:
But $sql is not about cleanup, but about creating the STIG schema. If that fails, maybe an old version for another database is used and would create wrong results.